lib/ansible/inventory/manager.py

```diff
@@ -161,13 +165,60 @@ def __init__(self, loader, sources=None, parse=True, cache=True):
         else:
             self._sources = sources
 
+        self._launch_ssh_agent()
+
         # get to work!
         if parse:
             self.parse_sources(cache=cache)
 
         self._cached_dynamic_hosts = []
         self._cached_dynamic_grouping = []
 
+    def _launch_ssh_agent(self):
+        ssh_agent_cfg = C.config.get_config_value('SSH_AGENT')
+        match ssh_agent_cfg:
+            case 'none':
+                display.debug('SSH_AGENT set to none')
+                return
+            case 'auto':
+                try:
+                    ssh_agent_bin = get_bin_path('ssh-agent', required=True)
+                except ValueError as e:
+                    raise AnsibleError('SSH_AGENT set to auto, but cannot find ssh-agent binary') from e
+                ssh_agent_dir = os.path.join(C.DEFAULT_LOCAL_TMP, 'ssh_agent')
+                os.mkdir(ssh_agent_dir, 0o700)
+                sock = os.path.join(ssh_agent_dir, 'agent.sock')
+                p = subprocess.Popen(
+                    [ssh_agent_bin, '-D', '-s', '-a', sock],
+                    stdin=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+                if p.poll() is not None:
+                    raise AnsibleError(
+                        f'Could not start ssh-agent: rc={p.returncode} stderr="{p.stderr}"'
+                    )
+                if (stdout := p.stdout.read(13)) != b'SSH_AUTH_SOCK':
+                    display.warning(
+                        f'The first 13 characters of stdout did not match the '
+                        f'expected SSH_AUTH_SOCK. This may not be the right binary, '
+                        f'or an incompatible agent: {stdout.decode()}'
+                    )
+                display.vvv(f'SSH_AGENT: ssh-agent[{p.pid}] started and bound to {sock}')
+                atexit.register(p.terminate)
+            case _:
+                sock = ssh_agent_cfg
+
+        try:
+            with SshAgentClient(sock) as client:
+                client.list()
+        except Exception as e:
```

Comment:
I'd be more comfortable with a known exception rather than shadowing arbitrary future problems.

Reply:
Is this really shadowing anything though? We don't want to handle the error, this just reraises `Exception` as `AnsibleError` and fails early on any error that is encountered on smoke testing the client.

Reply:
Such things should be done closer to the outer layer of a program, though. When the context is lost deep in the core, it's rather harmful. I'm hearing changes bundled with DT might help with this but haven't looked into those yet.

Reply:
does this belong in inventory_manger? seems more like something we should start at the specific CLIs. adhoc/playbook/console. For example ansible-inventory would still start the manager, but it has no good reason to do so.